src/clj/chronograph/handlers/timer.clj

```diff
@@ -0,0 +1,138 @@
+(ns chronograph.handlers.timer
+  (:require [chronograph.domain.timer :as timer]
+            [next.jdbc :as jdbc]
+            [chronograph.db.core :as db]
+            [chronograph.domain.task :as task]
+            [ring.util.response :as response]
+            [clojure.spec.alpha :as s]))
+
+(defn create
+  "Authorized users may create a timer for the give task id."
+  [{{:keys [task-id note] :as body} :body
+    {user-id :users/id} :user
+    :as _request}]
+  (jdbc/with-transaction [tx db/datasource]
+    (cond
+      (not (s/valid? :handlers.timer/create-request-body body))
+      (response/bad-request
+       {:error "Invalid Task ID or Note."})
+
+      (empty? (task/find-by-id tx task-id))
+      (response/bad-request
+       {:error "Task does not exist."})
+
+      :else (-> (timer/create! tx user-id task-id note)
+                response/response))))
+
+(defn delete
+  "Authorized users may delete any timer they own."
+  [{{:keys [timer-id]} :params
+    {user-id :users/id} :user
+    :as _request}]
+  (let [timer-id (timer/coerce-to-uuid timer-id)]
+    (jdbc/with-transaction [tx db/datasource]
+      (if (not (s/valid? :timers/id timer-id))
+        (response/bad-request
+         {:error "Invalid Timer ID."})
+        (if-let [time-span (timer/delete! tx user-id timer-id)]
+          (-> time-span
+              response/response)
+          (response/bad-request
+           {:error "Timer does not exist."}))))))
+
+(defn update-note
+  "Authorized users may update note for a timer they own, for the give task id."
+  [{{:keys [timer-id]} :params
+    {:keys [note]} :body
+    {user-id :users/id} :user
+    :as _request}]
+  (let [timer-id (timer/coerce-to-uuid timer-id)]
+    (jdbc/with-transaction [tx db/datasource]
+      (cond
+        (not (and (s/valid? :timers/id timer-id)
+                  (s/valid? :timers/note note)))
+        (response/bad-request
+         {:error "Invalid Timer ID or Note."})
+
+        (empty? (timer/find-for-user-by-timer-id tx user-id timer-id))
```

Comment:
Not worth changing right now, IMO. But, this could be done in the same query that's doing the update. Similar to how it's done in the delete query above.

Reply:
kept as-is